cpp/src/inflight_requests.cpp

```diff
@@ -21,11 +21,16 @@ void InflightRequests::insert(std::shared_ptr<Request> request)
   _inflightRequests->insert({request.get(), request});
 }
 
-void InflightRequests::merge(InflightRequestsMapPtr inflightRequestsMap)
+void InflightRequests::merge(InflightRequestsMapPtrPair inflightRequestsMapPtrPair)
 {
-  std::lock_guard<std::mutex> lock(_mutex);
-
-  _inflightRequests->merge(*inflightRequestsMap);
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    _inflightRequests->merge(*(inflightRequestsMapPtrPair.first));
```

Comment:
**question:** Now that these are `unique_ptr`s, should this be `std::move(inflightrequestsmapptrpair.first)` rather than `*...`?

Reply:
No, `merge` takes an object to merge from and not a pointer to the object. Move would be appropriate if we were instead purely replacing `_trackedRequest->_inflight` with it.

Reply:
OK, thanks!